controllers/front/callbacknotification.php

```diff
@@ -205,6 +213,13 @@ public function postProcess()
                         exit('Unrecognized status received ' . $transactionStatus);
                     }
                 }
+            } elseif (Validate::isLoadedObject($order) and $transactionStatus === 'epayment_declined') {
+                // Update payment status to 'declined'
+                $sql = 'UPDATE `' . _DB_PREFIX_ . $tableName . '`
+                        SET `paymentStatus` = \'declined\' WHERE `id_order` = ' . (int) $order->id;
```

Comment:
Casting to int can cause the result to be 0, which might be a valid order_id, causing side effects. You need to validate the integrity of the value instead. If you expect and INT, then validate the value is indeed an INT before casting.

Reply:
@emicha, we make sure the cart object and its ID are valid before using them in the code. Here's how we do it:

```
            // Load the cart
            $cart = getCartFromUniqueId($shopOrderId);
            if (!Validate::isLoadedObject($cart)) {
                exit('Could not load cart - exiting');
            }
```
This check ensures that cart->id is always valid, which prevents any potential issues with casting.